### DISCUSSION TITLE
Support for "Matplotlib" definitions wrapper kwargs in "colour.plotting" package.

### DISCUSSION THREAD

**Discussion by henczati** [colour-science/colour#677] [General]
Maybe in some future version you might consider using "wrapper" kwargs, and pass options to more frequently used matplotlib functions with unpacking tuples, lists, and dicts, with a special case when passing just a single parameter. E.g.:

``` python
some_plot(x, y,
          xlim=(None, 10), ylim=dict(ymax=200),
          xlabel="Single arg passed as single param to pyplot.xlabel()",
          legend=dict(ncol=3, pos="lower right", bbox_to_anchor=(0, 0.5)),
          ...)
```

PROS:
- Simple implementation: you just unpack and propagate to the appropriate function.
- No need to learn new plotting option names if one already knows `matplotlib`.
- You can access a lot of matplotlib functions with minimal effort.

CONS:
- If the single argument to be passed to a matplotlib function would be an unpackable type (tuple, list dict), it might feel weird and/or unintuitive that it would have to be wrapped once more.
- It would probably need (possibly major) refactoring to keep the code simple, and change the API.
- I might be just overthinking it... :)

**Comment by thomasmansencal** [comment]
It is actually a good idea and I was thinking about something along those lines following our discussion in #198.

I'll need to put more thinking into it though.

**Comment by amons000** [comment]
I hope this could be added. 
As a newbie, I want to learn matplotlib. Also, I would like this colour module could use the same kwargs that could be easier to people like me to learn both at the same time.
And I want to plot multiple SD in one figure but different axes, how could I make that? Please give me some hits.
Thanks~~~~

**Comment by thomasmansencal** [comment]
@Edwardlin-zlt : I'm a bit unclear on what you are asking for exactly, are you referring to sub-plots, e.g. https://matplotlib.org/api/_as_gen/matplotlib.pyplot.subplots.html?
